### PR TITLE
[Enhancement](external catalog) Added status reset when jdbc name mapping is abnormal

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

This PR introduces several important improvements aimed at enhancing the robustness, maintainability, and error handling efficiency of database connectivity and data retrieval processes. Here are the main optimization points:

1. **Code Abstraction and Reuse**:
   - Introduced `getRequiredMapping` method to centralize the common logic for retrieving database, table, and column names. This change reduces repetitive code across multiple functions, making the overall code structure more compact and consistent.

2. **Unified Error Handling**:
   - In the new `getRequiredMapping` method, missing mappings are handled uniformly. If a mapping is not found, the corresponding loading function is automatically triggered, and an exception is thrown if data remains missing. This approach simplifies error management and enhances code readability and maintainability.

3. **Enhanced Exception Safety**:
   - Added more rigorous checks at all critical data access points to ensure timely feedback and handling of exceptions when data is not successfully loaded, rather than just relying on the default behavior of data structures.

4. **Improved Robustness**:
   - Used `ConcurrentHashMap` and `AtomicBoolean` to ensure data consistency and thread safety in a multi-threaded environment. Optimized data checks and exception handling logic reduce potential issues caused by concurrent modifications.

5. **Improved Logging**:
   - Enhanced detailed logging at all points of data loading and exception handling. This not only helps in monitoring runtime states but also facilitates troubleshooting and subsequent debugging.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/IdentifierMapping.java [hunk 1]
```diff
@@ -179,51 +182,58 @@ public List<Column> setColumnNameMapping(String remoteDbName, String remoteTable
...
+        ConcurrentHashMap<String, ConcurrentHashMap<String, String>> tableColumnMap
+                = localColumnToRemoteColumn.computeIfAbsent(remoteDbName, k -> new ConcurrentHashMap<>());
+        Map<String, String> columnMap = tableColumnMap.computeIfAbsent(remoteTableName, k -> new ConcurrentHashMap<>());
+        if (columnMap.isEmpty()) {
+            LOG.info("Column name mapping missing, loading column names for localDbName: {}, localTableName: {}",
+                    localDbName, localTableName);
             loadColumnNamesIfNeeded(localDbName, localTableName);
         }
-        return localColumnToRemoteColumn.get(remoteDbName).get(remoteTableName);
+        if (columnMap.isEmpty()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/IdentifierMapping.java:208]: Even if you run `loadColumnNamesIfNeeded()`, the `columnMap` is still empty

File: fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/IdentifierMapping.java [hunk 2]
```diff
@@ -179,51 +182,58 @@ public List<Column> setColumnNameMapping(String remoteDbName, String remoteTable
...
 
+
     private void loadDatabaseNamesIfNeeded() {
         if (dbNamesLoaded.compareAndSet(false, true)) {
-            loadDatabaseNames();
+            try {
+                loadDatabaseNames();
+            } catch (Exception e) {
+                dbNamesLoaded.set(false); // Reset on failure
+                LOG.error("Error loading database names", e);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/IdentifierMapping.java:223]:
```suggestion
                LOG.warn("Error loading database names", e);
```

**Review by morningman — COMMENTED** [review]
Change to log level to `warn`

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Comment by zy-kkk** [comment]
run cloud_p1

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by kaka11chen — APPROVED** [review]
LGTM

**Pull request merged**